### PR TITLE
Fix merging volume claim templates

### LINKED ISSUE #145: How to adjust persistent volume claim size? (Type: ReplicaSet)
I looked at the YAML reference at https://docs.mongodb.com/kubernetes-operator/master/tutorial/deploy-replica-set/ and have been trying to tweak the following YAML:
```
apiVersion: mongodb.com/v1
kind: MongoDB
metadata:
  name: rocketchat-mdb
spec:
  members: 3
  type: ReplicaSet
  version: "4.2.6"
  persistent: true
  podSpec:
    memory: 512M
    persistence:
      multiple:
        data:
          storage: 1Gi
        journal:
          storage: 500M
        logs:
          storage: 500M
```
However, each member of the replica set still has 10Gi per PVC. How can I correctly change it?

### PR DESCRIPTION
**Pull request by chatton**
### All Submissions:

* [X] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

This PR replaces mergo with a manual merging of `PersistentVolumeClaims`. This PR also adds a mechanism for testing fixtures and verifying the StatefulSet was correctly constructed in unit tests.

closes #145

### CODE REVIEW

File: pkg/controller/mongodb/replica_set_controller.go [hunk 1]
```diff
@@ -197,6 +199,9 @@ func (r *ReplicaSetReconciler) Reconcile(request reconcile.Request) (reconcile.R
 
 	currentSts := appsv1.StatefulSet{}
 	if err := r.client.Get(context.TODO(), mdb.NamespacedName(), &currentSts); err != nil {
+		if apiErrors.IsNotFound(err) {
+			return reconcile.Result{}, err
+		}
 		r.log.Warnf("Error getting StatefulSet: %s", err)
 		return reconcile.Result{}, err
```
Comment (bznein) [inline pkg/controller/mongodb/replica_set_controller.go:206]: [nit] I would prefer the following:
```golang
if !apiErrors.IsNotFound(err) {
                 r.log.Warnf("Error getting StatefulSet: %s", err)
}	
return reconcile.Result{}, err			
```

Makes it easier for me to understand that the only difference is in the logging

File: pkg/controller/mongodb/testdata/change_data_volume.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  statefulSet:
+    spec:
+      volumeClaimTemplates:
+        - metadata:
+            name: data-volume
+          spec:
+            accessModes: [ "ReadWriteOnce", "ReadWriteMany" ]
+            resources:
+              requests:
+                storage: 50Gi
```
Comment (bznein) [inline pkg/controller/mongodb/testdata/change_data_volume.yaml:32]: I see github here warning me about no newline at the end of file. Do we care?

**Review by bznein — APPROVED** [review]
LGTM! two minor comments!

File: pkg/controller/mongodb/testdata/change_data_volume.yaml (hunk 2 above)
Comment (chatton) [reply pkg/controller/mongodb/testdata/change_data_volume.yaml:32]: I'll fix it!

File: pkg/controller/mongodb/replica_set_controller.go (hunk 1 above)
Comment (chatton) [reply pkg/controller/mongodb/replica_set_controller.go:206]: yep looks good to me!

**Pull request merged**